### ISSUE TITLE
can't pass arbitrary data to secho

### ISSUE THREAD

**Issue by kevindawson** [pallets/click#1680]
hi new be here just having a poke around - and came across this
```python
@cli.command()
@click.argument('src')
@click.argument('dest', required=False)
@pass_repo
def clone(repo, src, dest):
    click.secho(dir(repo), err=True, fg="cyan")
```

### Expected Behavior

Tell us what should happen.
```python
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'home', 'debug']
```

### Actual Behavior

```pytb
Traceback (most recent call last):
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/bin/pop3isp", line 33, in <module>
    sys.exit(load_entry_point('pop3isp', 'console_scripts', 'pop3isp')())
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/kevin/git/pop3isp/src/pop3isp/cli.py", line 44, in clone
    click.secho(dir(repo), err=True, fg="cyan")
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/termui.py", line 547, in secho
    message = style(message, **styles)
  File "/home/kevin/.local/share/virtualenvs/pop3isp-OYLenBlA/lib/python3.8/site-packages/click/termui.py", line 519, in style
    return "".join(bits)
TypeError: sequence item 1: expected str instance, list found
```

### Environment Ubuntu 20.04

* Python version: 3.8.2
* Click version: 7.1.2

ps. why do I have to do this
```python
click.secho(f"{dir(repo)}", err=True, fg="cyan")
```
thanks in advance

**Comment by davidism** [comment]
I'm not sure what the original title was referring to, this has nothing to do with interleaved commands, and they aren't present in the example either.

`secho` can only render strings, you're passing a list to it. However, that is addressed by #1146.

**Comment by kevindawson** [comment]
@davidism 
* thanks for the info
* sorry new-be issue - had you considered adding ```Call str() on any non-string type.``` to *Echoing* in documentation + example
* suggest you consider enabling *discussions* in github as you are losing lots of information (#1146)

_so long and thanks for all the fish!_
